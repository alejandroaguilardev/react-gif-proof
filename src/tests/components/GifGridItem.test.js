import GifGridItem from "../../components/GifGridItem";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

describe('Prueba de Componente', () => {
    const title = 'Un titulo';
    const url ="https://dominio.com/image";
    const wrapper = shallow(<GifGridItem title={ title } url= { url } />);

    test("Debe de mostrar el componte correctamente", () =>{
        expect(wrapper).toMatchSnapshot();
    }); 

    test('debe de tener un parrafo con el texto con el title', ()=>{
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('debe de tener la imagen igual a la url y alt', ()=>{
        const img = wrapper.find('img');
        // console.log(img.html())
        // console.log(img.prop())
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('debe de tener la clase animate', ()=>{
        const div = wrapper.find('div');
        const className  = div.prop('className');

        console.log(div.props())
        expect( className.includes('animate__fadeIn') ).toBe( true );
    });


});  