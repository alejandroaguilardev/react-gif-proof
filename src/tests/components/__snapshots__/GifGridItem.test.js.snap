// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de Componente Debe de mostrar el componte correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://dominio.com/image"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
